capstone/src/main/webapp/script.js

```diff
@@ -1,28 +1,145 @@
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+async function youtube() {
+  const id = document.getElementById('channelIdForAnalysis').value;
+  if (!id) {
+    return;
+  }
+  const response = await fetch('/youtube_servlet?channelId=' + id,);
+  const status = await response.json();
+  const comments = status;
+  const commentListElement = document.getElementById('comment-list');
+  commentListElement.innerHTML = '';
+  const att = await submitInput()
+  commentListElement.append(att);;
+  //commentListElement.append(att);
+  //for(let attribute in att){
+ //for (const attribute of att){
+   for(var i=0;i<att.length;i++){
+     const total =[];
+     //commentListElement.append("'" +  att[i] + "'");
+  for (let item in comments.items) {
+    const score = await callPerspective(comments.items[item].snippet.topLevelComment.snippet.textOriginal, "en", [att[i]]);
+    //commentListElement.appendChild(document.createElement('br'));
+      total.push(score);
+  }
+    totalSum = arrSum(total)
+    totalAvg = totalSum/comments.items.length;
+    commentListElement.append("AVERAGE " + att[i] + " : " + totalAvg);
+    commentListElement.appendChild(document.createElement('br'));
+  }
+  }
+
+
+const ATTRIBUTES_BY_LANGUAGE = {
+  'en': ['TOXICITY', 'SEVERE_TOXICITY', 'TOXICITY_FAST', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT', 'SEXUALLY_EXPLICIT', 'FLIRTATION'],
+  'es': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK_EXPERIMENTAL', 'INSULT_EXPERIMENTAL', 'PROFANITY_EXPERIMENTAL', 'THREAT_EXPERIMENTAL'],
+  'fr': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK_EXPERIMENTAL', 'INSULT_EXPERIMENTAL', 'PROFANITY_EXPERIMENTAL', 'THREAT_EXPERIMENTAL'],
+  'de': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT'],
+  'it': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT'],
+  'pt': ['TOXICITY', 'SEVERE_TOXICITY', 'IDENTITY_ATTACK', 'INSULT', 'PROFANITY', 'THREAT']
+};
+
+/** Collects the user's input and calls Perspective on it */
+async function submitInput() {
+  /*const textElement = document.getElementById('textForAnalysis');
+  if (!textElement) {
+    return;
+  }
+  const langElement = document.getElementById('languageForAnalysis');
+  if (!langElement) {
+    return;
+  }*/
+  const attributes = document.getElementById("available-attributes").getElementsByTagName('input');
+  const requestedAttributes = []
+  for (let attribute of attributes) {
+    if (attribute.checked == true) {
+      requestedAttributes.push(attribute.value);	
+    }	
+  }
+  return requestedAttributes;
+  //await callPerspective(textElement.value, langElement.value, requestedAttributes);
+}
```

Comment:
Considering that this function might end up having slightly different functionality from the `submitInput()` that we have for the text analyzer, it might be suitable to create a separate JavaScript file for the YouTube module so we don't have weird things happening on other webpages.

Reply:
addressed in "rename JS file and add descriptions'